### PR TITLE
Updated contract storage chapter

### PR DESCRIPTION
**Pull request by OkoliEvans**
Added Storing enums section to the chapter as directed here: https://github.com/cairo-book/cairo-book/issues/419

@enitrat Please review and give feedback

### CODE REVIEW

File: src/ch99-01-03-01-contract-storage.md [hunk 1]
```diff
@@ -86,6 +86,33 @@ For example, the storage layout for the `owner` variable of type `Person` will r
...
+
+Enums can be written to storage similarly to struct, by implementing the `store` trait for that enum.
+
+```rust, noplayground
+{{#rustdoc_include ../listings/ch06-enums-and-pattern-matching/no_listing_08_enum_storage/src/lib.cairo:enum_store}}
+```
+
+### Enums storage layout
+
+Enums are stored as serialized felt252 values on Starknet. The serialized felt252 values indicate the index of each variant in the enum definition, and if a variant has a defined type, that type will be serialized too.
```
Comment (enitrat) [inline src/ch99-01-03-01-contract-storage.md:100]: > The serialized felt252 values indicate the index of each variant in the enum definition, and if a variant has a defined type, that type will be serialized too.

How are types associated to variants stored? It is not very clear what you mean by "serialized" here. I think you can make it clearer how the index and the value are stored. Illustrating it with a table displaying storage addresses used might be helpful to grasp the representation in storage

File: listings/ch06-enums-and-pattern-matching/no_listing_08_enum_storage/src/lib.cairo [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+mod Enums {
+
+    #[storage]
+    struct Storage {
+        location: Direction,
+    }
+
+    // ANCHOR: enum_store
+    #[derive(Drop, starknet::Store)]
+    enum Direction {
```
Comment (enitrat) [inline listings/ch06-enums-and-pattern-matching/no_listing_08_enum_storage/src/lib.cairo:11]: the example lacks values associated to variants

File: src/ch99-01-03-01-contract-storage.md [hunk 3]
```diff
@@ -86,6 +86,33 @@ For example, the storage layout for the `owner` variable of type `Person` will r
...
+{{#rustdoc_include ../listings/ch06-enums-and-pattern-matching/no_listing_08_enum_storage/src/lib.cairo:enum_store}}
+```
+
+### Enums storage layout
+
+Enums are stored as serialized felt252 values on Starknet. The serialized felt252 values indicate the index of each variant in the enum definition, and if a variant has a defined type, that type will be serialized too. 
+In our example, `North` being the first variant in the enum will be serialized to `0x01`, `East` will be serialized to `0x02`, and this continues till the last variant.
+
+
+To read enum variants stored to `Storage` we call `read` function on the variant:
```
Comment (enitrat) [inline src/ch99-01-03-01-contract-storage.md:104]: remove this part

**Review by enitrat — COMMENTED** [review]
CI fails, can you format your code example and make sure it compiles?

edit: actually, it would be better to follow the editorial line and just modify the [contract used for the examples ](https://book.cairo-lang.org/ch99-01-03-00-a-deeper-dive-into-contracts.html)and add an enum; and refer to this enum in this section like we did for the `Person` struct.
___
Let's improve our name registry contract to enable two registration models: finite and infinite. 

```
enum RegistrationType {
    Finite: u64 (end timestamp)
    Inifinite
}
```

you will add a LegacyMap from `ContractAddress` to `RegistrationType` in Storage, and change `store_name` to take the `RegistrationType` as input; and write this type to storage.

**Comment by enitrat** [comment]
Hi @OkoliEvans let me know if you need help on this

**Comment by OkoliEvans** [comment]
Alright Sir, I am done with the corrections on the codes, it's remaining to fully understand the handle_enum implementation so I can complete the enum layout description @enitrat

**Comment by enitrat** [comment]
@OkoliEvans you can push the updated contract for an initial review if you want

**Comment by OkoliEvans** [comment]
Okay, will push now.

**Comment by OkoliEvans** [comment]
The new contract has been pushed @enitrat

File: src/ch99-01-03-01-contract-storage.md [hunk 4]
```diff
@@ -71,8 +71,12 @@ But what if you wanted to store a type that you defined yourself, such as an enu
 
 In our example, we want to store a `Person` struct in storage, which is possible by implementing the `Store` trait for the `Person` type. This can be achieved by simply adding a `#[derive(starknet::Store)]` attribute on top of our struct definition.
 
+## Storing Enums
+
+Enums can be written to storage similarly to struct, by implementing the `store` trait for that enum.
+
 ```rust, noplayground
-{{#rustdoc_include ../listings/ch99-starknet-smart-contracts/listing_99_03_example_contract/src/lib.cairo:person}}
```
Comment (enitrat) [inline src/ch99-01-03-01-contract-storage.md:75]: where did this snippet go?  Restore it above line 74

File: src/ch99-01-03-01-contract-storage.md [hunk 5]
```diff
@@ -71,8 +71,12 @@ But what if you wanted to store a type that you defined yourself, such as an enu
 
 In our example, we want to store a `Person` struct in storage, which is possible by implementing the `Store` trait for the `Person` type. This can be achieved by simply adding a `#[derive(starknet::Store)]` attribute on top of our struct definition.
 
+## Storing Enums
```
Comment (enitrat) [inline src/ch99-01-03-01-contract-storage.md:74]: No need for a specific subsection about enums here. You can just continue the previous paragraph which is generic over custom types, not specific to structs nor enums: 

> Similarly, we can store our `RegistrationType` enum...

File: src/ch99-01-03-01-contract-storage.md [hunk 6]
```diff
@@ -71,8 +71,12 @@ But what if you wanted to store a type that you defined yourself, such as an enu
...
+
+Enums can be written to storage similarly to struct, by implementing the `store` trait for that enum.
+
 ```rust, noplayground
-{{#rustdoc_include ../listings/ch99-starknet-smart-contracts/listing_99_03_example_contract/src/lib.cairo:person}}
+{{#rustdoc_include ../listings/ch99-starknet-smart-contracts/listing_99_03_example_contract/src/lib.cairo:enum_store}}
```
Comment (enitrat) [inline src/ch99-01-03-01-contract-storage.md:79]: and thus, this snippet can be removed as it's redundant.

**Review by enitrat — COMMENTED** [review]
I will let you apply the changes, let me know when you have come up with a clear explanation for the enum storage layout or if you need help writing it.

**Comment by OkoliEvans** [comment]
All changes suggested have been effected @enitrat , pls I will contact you on TG I need you assistance in fixing the enum layout description

File: src/ch99-01-03-01-contract-storage.md [hunk 7]
```diff
@@ -71,10 +71,12 @@ But what if you wanted to store a type that you defined yourself, such as an enu
 
 In our example, we want to store a `Person` struct in storage, which is possible by implementing the `Store` trait for the `Person` type. This can be achieved by simply adding a `#[derive(starknet::Store)]` attribute on top of our struct definition.
 
-```rust, noplayground
```
Comment (enitrat) [inline src/ch99-01-03-01-contract-storage.md:74]: fix this line

Comment (OkoliEvans) [reply src/ch99-01-03-01-contract-storage.md]: Done @enitrat

**Pull request merged**